src/main/java/com/google/sps/data/RestaurantChooser.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.sps.data.Response;
+import com.google.sps.data.Restaurant;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Random;
+
+public final class RestaurantChooser {
+    public static void chooseRestaurant(Response response, int requestedPrice){
+        if(response.getResults().size() == 0){
+            response.setStatus("ZERO_RESULTS");
+            return;
+        }
+        HashMap<String, Integer> restaurantScores = new HashMap<>();
```

Comment:
Maybe I am not understanding this algorithm correctly, but why not

Map<Integer, String> ( scores : Restaurant )
then randomly select an index from the size of the keySet
then output the associated value?